android-pay/src/main/java/com/stripe/wrap/pay/activity/StripeAndroidPayActivity.java

```diff
@@ -605,6 +632,85 @@ protected void onStripePaymentSourceReturned(
 
     /*------ End Overrides ------*/
 
+    private void getStripeSource(@NonNull final FullWallet fullWallet, @NonNull final Token token) {
+        mHandlerThread = new HandlerThread("Stripe networking");
+        mHandlerThread.start();
+        mStripeUiHandler = new Handler(Looper.getMainLooper()) {
+            private FullWallet wallet = fullWallet;
+            private Token originalToken = token;
+
+            @Override
+            public void handleMessage(Message msg) {
+                super.handleMessage(msg);
+                switch (msg.what) {
+                    case MSG_SOURCE_COMPLETE:
+                        Bundle bundle = msg.getData();
+                        mHandlerThread.quit();
+                        mHandlerThread = null;
+                        if (bundle != null) {
+                            String sourceString = bundle.getString("source", "shiiit");
+                            Source source = Source.fromString(sourceString);
+                            if (source != null && wallet != null) {
+                                logApiCallOnNewThread(source, null);
+                                onStripePaymentSourceReturned(wallet, source);
+                                return;
+                            }
+                        }
+
+                        logApiCallOnNewThread(originalToken, null);
+                        onStripePaymentSourceReturned(wallet, originalToken);
```

Comment:
I agree, the intention behind this is to have some option for users who are using sources to get sources everywhere (instead of something things sources and some things tokens). There should likely be some sort of configuration option that controls whether Android Pay results in sources or tokens, and whatever you set you always get that or null.